packages/android_alarm_manager/android/src/main/java/io/flutter/plugins/androidalarmmanager/BackgroundExecutionContext.java

```diff
@@ -0,0 +1,204 @@
+
+  public static void setPluginRegistrant(PluginRegistrantCallback callback) {
+    pluginRegistrantCallback = callback;
+  }
+
+  /**
+   * Sets the Dart callback handle for the Dart method that is responsible for initializing the
+   * background Dart isolate, preparing it to receive Dart callback tasks requests.
+   */
+  public static void setCallbackDispatcher(Context context, long callbackHandle) {
+    SharedPreferences prefs = context.getSharedPreferences(AlarmService.SHARED_PREFERENCES_KEY, 0);
+    prefs.edit().putLong(CALLBACK_HANDLE_KEY, callbackHandle).apply();
+  }
+
+  /** Returns true when the background isolate has started. */
+  public boolean isRunning() {
+    return isIsolateRunning.get();
+  }
+
+  private void onInitialized() {
+    isIsolateRunning.set(true);
+    AlarmService.onInitialized();
+  }
+
+  @Override
+  public void onMethodCall(MethodCall call, Result result) {
+    String method = call.method;
+    Object arguments = call.arguments;
+    try {
+      if (method.equals("AlarmService.initialized")) {
+        // This message is sent by the background method channel as soon as the background isolate
+        // is running. From this point forward, the Android side of this plugin can send
+        // callback handles through the background method channel, and the Dart side will execute
+        // the Dart methods corresponding to those callback handles.
+        onInitialized();
+        result.success(true);
+      } else {
+        result.notImplemented();
+      }
+    } catch (PluginRegistrantException e) {
+      result.error("error", "AlarmManager error: " + e.getMessage(), null);
+    }
+  }
+
+  public void startBackgroundIsolate(Context context) {
+    if (!isRunning()) {
+      SharedPreferences p = context.getSharedPreferences(AlarmService.SHARED_PREFERENCES_KEY, 0);
+      long callbackHandle = p.getLong(CALLBACK_HANDLE_KEY, 0);
+      startBackgroundIsolate(context, callbackHandle);
+    }
+  }
+
+  /**
+   * Starts running a background Dart isolate within a new {@link FlutterEngine}.
+   *
+   * <p>The isolate is configured as follows:
+   *
+   * <ul>
+   *   <li>Bundle Path: {@code FlutterMain.findAppBundlePath(context)}.
+   *   <li>Entrypoint: The Dart method represented by {@code callback}.
+   *   <li>Run args: none.
+   * </ul>
+   *
+   * <p>Preconditions:
+   *
+   * <ul>
+   *   <li>The given {@code callback} must correspond to a registered Dart callback. If the handle
+   *       does not resolve to a Dart callback then this method does nothing.
+   *   <li>A static {@link #pluginRegistrantCallback} must exist, otherwise a {@link
+   *       PluginRegistrantException} will be thrown.
+   * </ul>
+   */
+  public void startBackgroundIsolate(Context context, long callbackHandle) {
+    if (backgroundFlutterEngine != null) {
+      Log.e(TAG, "Background isolate already started");
+      return;
+    }
+
+    FlutterMain.ensureInitializationComplete(context, null);
```

Comment:
On master this call should no longer be needed.

Reply:
Is this still required for apps using the V1 embedding?

Reply:
Yeah I think so. In short, if a Flutter experience is based on a `FlutterEngine` then that call is not needed. But if it's based on `FlutterNativeView` then it is.